### ISSUE TITLE
HEIC pictures get downloaded instead of viewed in files

### ISSUE THREAD

**Issue by tiiiecherle** [nextcloud/server#15274]
Hey,

first of all thanks for alle the hard work put into this. Keep it up, very much appreciated.

When logged in via Safari on macOS (same happens with Firefox) in your nextcloud account and you upload a .jpg and a .heic to a folder then when clicking on the .jpg the picture viewer opens and displays the picture. When clicking on the .heic the picture gets downloaded instead of getting viewed in the browser.

Thumbnail preview works for both, .jpg and .heic and both are shown correctly in the gallery. The .heic can be shown by clicking gallery and then clicking on the .heic picture.

So only the click on the .heic picture in files in the browser seems to have a wrong action associated and should view the .heic picture instead of downloading.

- nextcloud version 16.0.0
- nginx on archlinux with all packages up-to-date
- preview is enabled for .heic files ('OC\Preview\HEIC',)
- libheif and imagemagick are installed
- php-imagick is installed and activated
- tested on macos 10.14.4 Safari and Firefox

Thanks for looking into it in advance.

**Comment by alsacefan** [comment]
I have exactly the same issue.  NC version 16.0.0.

**Comment by schmurtzm** [comment]
Hi, Same behavior here with NC 17.0.2 .  I run NextCloud in a Nextcloud VM Pre-installed (ubuntu 18.04.3 , NextCloud 17.0.2)

I've followed [this article](https://medium.com/@eplt/5-minutes-to-install-imagemagick-with-heic-support-on-ubuntu-18-04-digitalocean-fe2d09dcef1) to compile ImageMagick with HEIC support and update "PHP Imagick Plugin" and now I've got thumbnails on my HEIC files.

So HEIC thumbnails are OK but still not possible to see it in the viewer. For example if I open a .jpg file and then I use right arrow to see the next image file, I see only .jpg files and not the HEIC : these are bypassed by the viewer.
